src/composables/useArrowNavigation.js

```diff
@@ -49,16 +49,20 @@ export function useArrowNavigation(listElementRef, defaultElementRef, selector,
 
 	// Set focused index according to selected element
 	const handleFocusEvent = (event) => {
-		const newIndex = itemElementsIdMap.value.findIndex(id => id === event.target?.id)
+		const newIndex = itemElementsIdMap.value.indexOf(event.target?.id)
 
-		if (newIndex !== -1) {
+		// Quit if triggered by arrow navigation as already handled
+		// or if using Tab key to navigate, and going through NcActions
+		if (focusedIndex.value !== newIndex && newIndex !== -1) {
 			focusedIndex.value = newIndex
 		}
 	}
 
-	// Reset focused index if focus moved out of navigation area
+	// Reset focused index if focus moved out of navigation area or moved to the defaultRef
 	const handleBlurEvent = (event) => {
-		if (!listRef.value.contains(event.relatedTarget)) {
+		if (!listRef.value.contains(event.relatedTarget)
+			|| defaultRef.value?.$el.contains(event.relatedTarget)
+			|| defaultRef.value.contains?.(event.relatedTarget)) {
```

Comment:
I am not sure if I understand correctly your assumptions here 
because `defaultRef` is always assumed to be a DOM element 
```
		// depending on ref, listElementRef could be either a component or a DOM element
		listRef.value = unref(listElementRef)?.$el ?? unref(listElementRef)
		defaultRef.value = unref(defaultElementRef)
```

Reply:
That was important for `leftSidebar`, as we're using it only as DOM element in the code

`defaultRef`, on the other hand, could be either a DOM element or a Vue component, so we should check here both options:
- DOM doesn't have `$el` prop
- Vue component doesn't have `contain()` method